tests/ui/features/@upload/sbom-upload.step.ts

```diff
@@ -0,0 +1,126 @@
+import fs from "fs";
+import path from "path";
+import { expect, test } from "@playwright/test";
+import { createBdd } from "playwright-bdd";
+
+export const { Given, When, Then, Step } = createBdd();
+
+const MENU_UPLOAD = "Upload";
+const HEADER_UPLOAD = "Upload";
+const BUTTON_UPLOAD = "Upload";
+const TAB_SBOM = "SBOM";
+const DRAG_INSTRUCTIONS = "Drag and drop files here or";
+const ACCEPTED_TYPES_DESC = "Accepted file types:";
+
+const TIMEOUT_PAGELOAD_IMPORT = 2_000;
+const TIMEOUT_UPLOAD_DONE = 90_000;
+
+const SBOM_SET = {
+  "single SBOM": ["test-upload-001.spdx.json"],
+  "multiple SBOMs": [
+    "test-upload-002.spdx.json",
+    "test-upload-003.spdx.json.bz2",
+    "test-upload-004.cdx.json",
+    "test-upload-005.cdx.json",
+    "test-upload-006.cdx.json.bz2",
+  ],
+};
+
+const visitUploadPage = async ({ page }) => {
+  await page.goto("/importers"); // dont care which page here, importers is lot faster than Dashboard
+  await page.getByRole("link", { name: MENU_UPLOAD }).click();
+
+  const header = page.locator(`xpath=(//h1[text()="${HEADER_UPLOAD}"])`);
+  await header.waitFor({ state: "visible", timeout: TIMEOUT_PAGELOAD_IMPORT });
+};
+Step("User visits Upload page", visitUploadPage);
+
+function assetsPath(files: string[]): string[] {
+  return files.map((e) => path.join(__dirname, "assets", e));
+}
+
+// PAGE CONTENT
+
+Then("SBOM upload tab is selected", async ({ page }) => {
+  await expect(page.getByRole("tab", { name: TAB_SBOM })).toHaveAttribute(
+    "aria-selected",
+    "true"
+  );
+});
+
+Then("Drag and drop instructions are visible", async ({ page }) => {
+  await expect(
+    page.getByLabel("SBOM").getByText(DRAG_INSTRUCTIONS)
+  ).toBeVisible();
+});
+
+Then("Upload button is present", async ({ page }) => {
+  await expect(page.getByRole("button", { name: "Upload" })).toBeVisible();
+});
+
+Then("Accepted file types are described", async ({ page }) => {
+  expect(page.getByLabel("SBOM").getByText(ACCEPTED_TYPES_DESC)).toBeVisible();
+});
+
+// FILE UPLOAD
+
+When("User uploads {string}", async ({ page }, data_set_key) => {
```

Comment:
Another soft suggestion - it would be easier to read without having to scroll up if instead of `data_set_key` the parameter was called something like `option` or `sbom_subset`.